src/test/java/de/hpi/bpt/argos/api/EventQueryEndpointTest.java

```diff
@@ -0,0 +1,244 @@
+		String newQuery = testQuery.getQuery() + "_new";
+
+		int eventQueriesCount = getEventQueriesCount();
+		WebSocket webSocket = WebSocket.buildWebSocket();
+
+		RestRequest request = RestRequestFactoryImpl.getInstance().createPutRequest(ARGOS_REST_HOST, getEditEventQueryUri(testQuery.getId()));
+
+		JsonObject newQueryJson = new JsonObject();
+		newQueryJson.addProperty("Description", newDescription);
+		newQueryJson.addProperty("Query", newQuery);
+
+		request.setContent(serializer.toJson(newQueryJson));
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		assertEquals(eventQueriesCount, getEventQueriesCount());
+
+		EventQuery updatedQuery = PersistenceAdapterImpl.getInstance().getEventQuery(testQuery.getId());
+		assertEquals(testQuery.getTypeId(), updatedQuery.getTypeId());
+		assertEquals(newDescription, updatedQuery.getDescription());
+		assertEquals(newQuery, updatedQuery.getQuery());
+
+		List<String> webSocketMessages = webSocket.awaitMessages(1, 1000);
+		ArgosTestUtil.assertWebSocketMessage(webSocketMessages.get(0), PersistenceArtifactUpdateType.MODIFY, "EventQuery");
+
+		JsonObject notification = jsonParser.parse(webSocketMessages.get(0)).getAsJsonArray().get(0).getAsJsonObject();
+		assertEquals(testQuery.getId(), notification.get("ArtifactId").getAsLong());
+	}
+
+	@Test
+	public void testEditEventQuery_InvalidEventQueryId_BadRequest() {
+		EventQuery testQuery = ArgosTestUtil.createEventQuery(testEventType, true);
+
+		String newDescription = testQuery.getDescription() + "_new";
+		String newQuery = testQuery.getQuery() + "_new";
+
+		int eventQueriesCount = getEventQueriesCount();
+
+		RestRequest request = RestRequestFactoryImpl.getInstance().createPutRequest(ARGOS_REST_HOST, getEditEventQueryUri(0));
+
+		JsonObject newQueryJson = new JsonObject();
+		newQueryJson.addProperty("Description", newDescription);
+		newQueryJson.addProperty("Query", newQuery);
+
+		request.setContent(serializer.toJson(newQueryJson));
+		assertEquals(HttpStatusCodes.BAD_REQUEST, request.getResponseCode());
+
+		assertEquals(eventQueriesCount, getEventQueriesCount());
+
+		EventQuery updatedQuery = PersistenceAdapterImpl.getInstance().getEventQuery(testQuery.getId());
+		assertEquals(testQuery.getTypeId(), updatedQuery.getTypeId());
+		assertEquals(testQuery.getDescription(), updatedQuery.getDescription());
+		assertEquals(testQuery.getQuery(), updatedQuery.getQuery());
+	}
+
+	@Test
+	public void testEditEventQuery_InvalidJson_BadRequest() {
+		EventQuery testQuery = ArgosTestUtil.createEventQuery(testEventType, true);
+
+		String newQuery = testQuery.getQuery() + "_new";
+
+		int eventQueriesCount = getEventQueriesCount();
+
+		RestRequest request = RestRequestFactoryImpl.getInstance().createPutRequest(ARGOS_REST_HOST, getEditEventQueryUri(0));
+
+		JsonObject newQueryJson = new JsonObject();
+		// remove this attribute to make the json invalid
+		//newQueryJson.addProperty("Description", newDescription);
```

Comment:
Maybe just have a comment that the description attribute is not included without actualling commenting the original line.